src/frame.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+function Frame() {
+  this.rolls = [];
+  this.score = 0;
+  this.strike = false;
+  this.spare = false;
+}
+
+Frame.prototype.bowl = function(pins) {
+  if (this.score + pins > 10) {
+    throw new Error('Please re-enter correct score');
+  }
+  this.rolls.push(pins);
+  this.score += pins;
+  this.assignBonus();
+};
+
+Frame.prototype.isEndFrame = function() {
```

Comment:
`isEndFrame` says to me that this function tells me whether this frame is the last frame in the game, rather than that this frame is over.  Just `isOver` would be much more intuitive.